analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Common/Checks/ByteArrayCheck.cs

```diff
@@ -42,6 +44,16 @@ programPoint.CurrentInstruction switch
                 ? programState.SetConstraint(programState.PeekValue(), ByteArraySymbolicValueConstraint.Constant)
                 : programState;
 
+        private ProgramState ImplicitlyTypedArrayPostProcess(InitializerExpressionSyntax initializerExpression, ProgramState programState) =>
+            initializerExpression.IsKind(SyntaxKind.ArrayInitializerExpression)
```

Comment:
it's a bit difficult to realize which condition is not met by the coverage

Reply:
I wonder how coverage got to 9/10. By brute force measuring, I get 7/10.
These conditions are never `false`:
`initializerExpression.IsKind(SyntaxKind.ArrayInitializerExpression)`
`initializerExpression.FirstAncestorOrSelf<VariableDeclarationSyntax>() is VariableDeclarationSyntax variableDeclaration`
`programState.HasValue`

Reply:
I am puzzled, I added a test with CollectionInitialzer and it's still at 9/10
![image](https://user-images.githubusercontent.com/38876598/122236347-bc1c2000-cebe-11eb-9cca-b61e4895043c.png)